test/fk_ik_tests.cpp

```diff
@@ -0,0 +1,225 @@
+// Author: Jeroen Demaeyer(JeroenDM @ Github)
+// With edits by Jonathan Meyer
+
+#include <array>
+#include <chrono>
+#include <gtest/gtest.h>
+#include <random>
+
+#include "opw_kinematics/opw_kinematics.h"
+#include "opw_kinematics/opw_parameters_examples.h"
+#include "opw_kinematics/opw_utilities.h"
+
+
+// Structure used for setting sampler parameters based on floating point type
+template <typename T>
+struct TestTolerance;
+
+template<>
+struct TestTolerance<float> {
+  static constexpr float TOLERANCE = 1e-4f;
+  static constexpr const char* const NAME = "float";
+};
+
+template<>
+struct TestTolerance<double> {
+  static constexpr double TOLERANCE = 1e-10;
+  static constexpr const char* const NAME = "double";
+};
+
+// Helper Functions for generating and testing random poses in cartesian/joint space
+using opw_kinematics::Transform;
+
+template <typename T>
+void comparePoses(const Transform<T>& Ta, const Transform<T>& Tb, double tolerance)
+{
+  using Matrix = Eigen::Matrix<T, 3, 3>;
+  using Vector = Eigen::Matrix<T, 3, 1>;
+
+  Matrix Ra = Ta.rotation(), Rb = Tb.rotation();
+  for (int i = 0; i < Ra.rows(); ++i)
+  {
+    for (int j = 0; j < Ra.cols(); ++j)
+    {
+      EXPECT_NEAR(Ra(i, j), Rb(i, j), tolerance);
+    }
+  }
+
+  Vector pa = Ta.translation(), pb = Tb.translation();
+  EXPECT_NEAR(pa[0], pb[0], tolerance);
+  EXPECT_NEAR(pa[1], pb[1], tolerance);
+  EXPECT_NEAR(pa[2], pb[2], tolerance);
+}
+
+template <typename T>
+void getRandomJointValues(T* q)
+{
+  static std::default_random_engine en{ 42 };                  // random engine
+  static std::uniform_real_distribution<T> rg{ T(-M_PI), T(M_PI) };  // random generator
+
+  q[0] = rg(en);
+  q[1] = rg(en);
+  q[2] = rg(en);
+  q[3] = rg(en);
+  q[4] = rg(en);
+  q[5] = rg(en);
```

Comment:
Just out of curiosity, do you know a more elegant way to do this assignment of the joint values? I tried some kind of in-place initialization at the return statement but I couldn't figure out how.
(There is probably also no point in making it syntactically more complicated...apart from a for loop maybe.)